synapse/replication/tcp/redis/connection.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from typing import TYPE_CHECKING, Any, Generic, Optional, Type, TypeVar, cast
+
+import attr
+from txredisapi import (  # type: ignore[attr-defined]
+    ConnectionHandler,
+    RedisFactory,
+    Sentinel,
+)
```

Comment:
`txredis` is an optional dependency which now gets imported unconditionally, so this causes some tests to fail.

I'm not sure *where* to put the `try-catch`, however. Maybe back in synapse/replication/tcp/external_cache.py?